### PR TITLE
light-clients: make prefix empty for both cf-solana and cf-guest

### PR DESCRIPTION
**Pull request by dhruvja**
Since prefix is expected to be empty, send empty prefixes when verifying proofs.

### CODE REVIEW

**Comment by mina86** [comment]
Yes, but the point is that the prefix sent from relayer is empty and we want to verify that.  If it’s not empty than relayer sends wrong proof.

**Comment by dhruvja** [comment]
The old ibc doesnt support empty prefixes but i guess we can change it.
https://github.com/ComposableFi/composable-ibc/blob/fa46171b7972d38de5da8b29a999420fe2e580cb/ibc/modules/src/core/ics23_commitment/commitment.rs#L157

**Comment by mina86** [comment]
You can create them via `CommitmentPrefix::default()`.  It’s only when you try to create it from a vector that it complains.

**Comment by dhruvja** [comment]
oh ya true, didnt notice that. In that case we can close this PR

**Comment by dhruvja** [comment]
@mina86 apparently the old ibc doesnt support empty prefix at all.
I get the following error
```
 error converting message type into domain type: ics02 client error

    Caused by:
        empty prefix
```

File: common/cf-guest/src/client/impls.rs [hunk 1]
```diff
@@ -199,15 +199,15 @@ impl<PK: PubKey> ibc::ClientStateCommon for ClientState<PK> {
...
         &self,
-        prefix: &ibc::CommitmentPrefix,
+        _prefix: &ibc::CommitmentPrefix,
         proof: &ibc::CommitmentProofBytes,
         root: &ibc::CommitmentRoot,
         path: ibc::path::Path,
         value: Vec<u8>,
     ) -> Result {
         let value = Some(value.as_slice());
         proof::verify_for_block(
```
Comment (mina86) [inline common/cf-guest/src/client/impls.rs:209]:
```suggestion
        // TODO: Once IBC is updated everywhere to version which supports empty
        // prefixes, change `&[]` with `prefix.as_bytes()`.
        proof::verify_for_block(
```

**Review by mina86 — APPROVED** [review]
_sighs_ Why is IBC so broken? I guess alternative would be to patch ibc in composable-ibc.

**Pull request merged**